src/main/java/ca/bc/gov/backendstartapi/endpoint/FundingSourceEndpoint.java

```diff
@@ -0,0 +1,67 @@
+package ca.bc.gov.backendstartapi.endpoint;
+
+import ca.bc.gov.backendstartapi.entity.FundingSource;
+import ca.bc.gov.backendstartapi.repository.FundingSourceRepository;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.time.LocalDate;
+import java.util.List;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/** This class exposes funding sources resources API. */
+@Setter
+@RestController
+@NoArgsConstructor
+@RequestMapping("/api/funding-sources")
+@Tag(
+    name = "FundingSourceEndpoint",
+    description = "Resource to retrieve Funding Source to Owners Agencies")
+public class FundingSourceEndpoint {
+
+  private FundingSourceRepository fundingSourceRepository;
+
+  @Autowired
+  public FundingSourceEndpoint(FundingSourceRepository fundingSourceRepository) {
+    this.fundingSourceRepository = fundingSourceRepository;
+  }
+
+  /**
+   * Retrieve all funding sources.
+   *
+   * @return A list of {@link FundingSource} with all found result.
+   */
+  @GetMapping(produces = "application/json")
+  @PreAuthorize("hasRole('user_read')")
+  @Operation(
+      summary = "Retrieve all funding sources",
+      description = "Retrieve all valid (not expired) funding source")
```

Comment:
Summary and description don't match here: we state that the endpoint will return all the funding sources and later, when detailing the endpoint, that we're not actually returning all of them.

The summary could state that we'll return all valid funding sources and the description could state that *and* determine what "valid" means: sources that have already been put in effect (today ≥ `effectiveDate`) and haven't expired yet (today < `expiryDate`).

PS I'm awaiting a response from Sabina to see if `effectiveDate` ≤ today < `expiryDate` is indeed the right criteria of validity.

Reply:
Sabina confirmed that `effectiveDate` ≤ today < `expiryDate` is the right criterion.